site/source/docs/contracts_configuration.md

```diff
@@ -186,6 +186,30 @@ contracts: {
 ...
 ```
 
+## Dynamic Addresses
+
+There are scenarios in which we want to configure a Smart Contract that is already deployed by a third-party, but its address can only be computed at run-time. For such cases, Embark supports specifying an `addressHandler` which returns the address we're interested in. Usually, other Smart Contract instances are needed to perform that computation, so the [`deps` configuration](#Deployment-hooks) comes in handy as well:
+
+```
+contracts: {
+  SimpleStorage: {
+    fromIndex: 0,
+    args: [100],
+  },
+  OtherContract: {
+    deps: ['SimpleStorage'],
+    addressHandler: async (deps) => {
+      // use `deps.contracts.SimpleStorage` to determine address
```

Comment:
I'd specify here that they need to return the address at the end (it,s obvious, but we are never too clear 😉 )

Reply:
Good point.